src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -2276,8 +2276,13 @@ public static SSTableReader moveAndOpenSSTable(ColumnFamilyStore cfs, Descriptor
             throw new RuntimeException(msg);
         }
 
-        logger.info("Renaming new SSTable {} to {}", oldDescriptor, newDescriptor);
-        SSTableWriter.rename(oldDescriptor, newDescriptor, components);
+        if (copyData) {
+            logger.info("Copying new SSTable {} to {}", oldDescriptor, newDescriptor);
+            SSTableWriter.copy(oldDescriptor, newDescriptor, components);
```

Comment:
We should probably try to hard link the file here? If it fails we can fall back to copy